File: 0001_0100/0017/0017.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    p = f.readline().split()
    i = 1
    while i < n:
        if (n - 1) % i == 0 and p[i] == p[0]:
            j = i
            while j < n:
                if p[j] != p[j % i]:
                    break
                j += 1
            if j == n:
                break
        i += 1
    q.write(str(i))

File: 0001_0100/0017/0017.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, i, j int
	fmt.Fscan(f, &n)
	p := make([]int, n)
	for ; i < n; i++ {
		fmt.Fscan(f, &p[i])
	}
	for i = 1; i < n; i++ {
		if (n-1)%i == 0 && p[i] == p[0] {
			for j = i; j < n; j++ {
				if p[j] != p[j%i] {
					break
				}
			}
			if j == n {
				break
			}
		}
	}
	fmt.Fprint(q, i)
}
